src/components/v2/V2Project/V2PayButton/V2ConfirmPayModal/index.tsx

```diff
@@ -106,10 +101,26 @@ export default function V2ConfirmPayModal({
     }
     setLoading(true)
 
+    const imageUrls: string[] = []
+    let memoMetadata: Record<string, unknown> = { version: 1 }
+    if (uploadedImage) {
+      memoMetadata = { ...memoMetadata, userUploadedImages: [uploadedImage] }
+      imageUrls.push(uploadedImage)
+    }
+    if (stickerUrls) {
+      memoMetadata = { ...memoMetadata, stickerUrls }
+    }
+
+    const memo =
+      (textMemo ?? '') +
+      (Object.keys(memoMetadata).length
+        ? '\n' + JSON.stringify(memoMetadata)
+        : '')
+
```

Comment:
**thought**: I know we talked about this, but im wondering if this is overkill for now.

WDYT about appending stickers and the image to the memo instead, with no JSON?

```
<memo text> <sticker urls> <uploaded image>
```

With this, we can still differentiate the uploaded image from stickers when we re-work the payment activity element.

Reply:
I don't mind - reasoning behind this (especially with the version flag) is it allows us to update the format later and still provide backward compatibility

Reply:
I think let's roll with the simplest, boring solution. + it saves characters, saving a touch of gas ⛽️.

Do you foresee us having to update the format with any regular frequency?

Reply:
True true, hmm... I think the trouble with not having any way to control the format now means we cannot change it later in the future. What happens if we want to allow more expression in the memo? I dunno, wdyt? It buys us more flexibility for a trivial gas increase. If you want it looking better on chain, we could base64 encode it? ;)

Reply:
I don't feel strongly. Want to avoid over-engineering and opt for the most boring solution in general IMO.

Let's ship this as-is 👍